domains/pallets/executor-registry/src/lib.rs

```diff
@@ -768,6 +782,10 @@ mod pallet {
         ValueQuery,
     >;
 
+    /// Merkle root of authorities.
+    #[pallet::storage]
+    pub(super) type AuthoritiesRoot<T> = StorageValue<_, Blake2b256Hash, ValueQuery>;
```

Comment:
Do we plan to adopt this optimization for the core domain? Like introducing `DomainAuthoritiesRoot` to the `domain-registry` pallet.

Reply:
Actually, it's not needed on the core domain side due to the difference in the underlying storage.

- system domain: the entire executor set is tracked using a bounded vector https://github.com/subspace/subspace/blob/3fef2ef3b69e651b8185d44bc2cbf6553696ce60/domains/pallets/executor-registry/src/lib.rs#L779-L783
- core domain: core domain authorities (partial executor set) are tracked using a double map https://github.com/subspace/subspace/blob/3fef2ef3b69e651b8185d44bc2cbf6553696ce60/domains/pallets/domain-registry/src/lib.rs#L150-L158

I previously chose the bounded vector for the executor set because we aim to support 100K executors, and polkadot also configures the max authorities of babe to 100K, so it should be performance-wise okay within the runtime. Using the map doesn't have the concern of proof size, but the map needs to be cleared first and then be inserted with new values on epoch change, not sure it's okay for the entire executor set but seemingly fine for the domain authorities as it's a subset of the executors.

In all, the existing implementation is based on experience/intuition without precise measuring. We'll measure how they work actually, a TODO is already marked there, we'll revisit when the weight system is fixed. cc @vedhavyas